### ISSUE TITLE
ORA-00205: error in identifying control file, check alert log for more info

### ISSUE THREAD

**Issue by mlemnian** [gvenzl/oci-oracle-xe#157]
Everytime I want to start a *-faststart container with mounting the database folder from the host system (persistent container) I run into a `ORA-00205: error in identifying control file, check alert log for more info` error.

How to reproduce:
```
docker run -d -p 1521:1521 -e ORACLE_PASSWORD=password -v /temp/db/data:/opt/oracle/oradata gvenzl/oracle-xe:21.3.0-slim-faststart
```

will result into the following error:

```
CONTAINER: starting up...
CONTAINER: first database startup, initializing...
CONTAINER: starting up Oracle Database...

LSNRCTL for Linux: Version 21.0.0.0.0 - Production on 14-NOV-2022 14:13:58

Copyright (c) 1991, 2021, Oracle.  All rights reserved.

Starting /opt/oracle/product/21c/dbhomeXE/bin/tnslsnr: please wait...

TNSLSNR for Linux: Version 21.0.0.0.0 - Production
System parameter file is /opt/oracle/homes/OraDBHome21cXE/network/admin/listener.ora
Log messages written to /opt/oracle/diag/tnslsnr/acda3e76ae10/listener/alert/log.xml
Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC_FOR_XE)))
Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))

Connecting to (DESCRIPTION=(ADDRESS=(PROTOCOL=IPC)(KEY=EXTPROC_FOR_XE)))
STATUS of the LISTENER
------------------------
Alias                     LISTENER
Version                   TNSLSNR for Linux: Version 21.0.0.0.0 - Production
Start Date                14-NOV-2022 14:13:58
Uptime                    0 days 0 hr. 0 min. 0 sec
Trace Level               off
Security                  ON: Local OS Authentication
SNMP                      OFF
Default Service           XE
Listener Parameter File   /opt/oracle/homes/OraDBHome21cXE/network/admin/listener.ora
Listener Log File         /opt/oracle/diag/tnslsnr/acda3e76ae10/listener/alert/log.xml
Listening Endpoints Summary...
  (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC_FOR_XE)))
  (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))
The listener supports no services
The command completed successfully
ORACLE instance started.

Total System Global Area 1241512272 bytes
Fixed Size                  9685328 bytes
Variable Size             654311424 bytes
Database Buffers          570425344 bytes
Redo Buffers                7090176 bytes
ORA-00205: error in identifying control file, check alert log for more info
```

I think this is expected behavior because on a first boot, usually an empty host folder is mounted into the (already extracted DB) container folder, which breaks everything.

But, I think the documentation should highlight this behavior.

**Comment by gvenzl** [comment]
Hi @mlemnian,

Yes, you are correct, the error basically means that the process cannot find the database files, the control file being the first one to read to get information about all the others. Given that you mounted in an empty volume, the database is gone.

This is something that I have disliked about Docker from the beginning and to my disappointment Podman doesn't do it any different from what I just saw.
The behavior here highly depends on whether you are using just a name for the volume or a path. I think they called these *named volumes* and *host-only volumes* respectively back in the days, perhaps still now.

If you are using just a name, the runtime will first copy anything in the path that is mounted into that volume, so you will find that this works just perfectly well:

```
[gvenzl@localhost oci-oracle-xe]$ podman volume ls
[gvenzl@localhost oci-oracle-xe]$ podman run --name test -v test:/opt/oracle/oradata -e ORACLE_RANDOM_PASSWORD=y gvenzl/oracle-xe:21.3.0-slim-faststart
CONTAINER: starting up...
CONTAINER: first database startup, initializing...
CONTAINER: starting up Oracle Database...

LSNRCTL for Linux: Version 21.0.0.0.0 - Production on 03-DEC-2022 18:20:23

Copyright (c) 1991, 2021, Oracle.  All rights reserved.

Starting /opt/oracle/product/21c/dbhomeXE/bin/tnslsnr: please wait...

TNSLSNR for Linux: Version 21.0.0.0.0 - Production
System parameter file is /opt/oracle/homes/OraDBHome21cXE/network/admin/listener.ora
Log messages written to /opt/oracle/diag/tnslsnr/50c499a2992e/listener/alert/log.xml
Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC_FOR_XE)))
Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))

Connecting to (DESCRIPTION=(ADDRESS=(PROTOCOL=IPC)(KEY=EXTPROC_FOR_XE)))
STATUS of the LISTENER
------------------------
Alias                     LISTENER
Version                   TNSLSNR for Linux: Version 21.0.0.0.0 - Production
Start Date                03-DEC-2022 18:20:24
Uptime                    0 days 0 hr. 0 min. 0 sec
Trace Level               off
Security                  ON: Local OS Authentication
SNMP                      OFF
Default Service           XE
Listener Parameter File   /opt/oracle/homes/OraDBHome21cXE/network/admin/listener.ora
Listener Log File         /opt/oracle/diag/tnslsnr/50c499a2992e/listener/alert/log.xml
Listening Endpoints Summary...
  (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC_FOR_XE)))
  (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))
The listener supports no services
The command completed successfully
ORACLE instance started.

Total System Global Area 1241512272 bytes
Fixed Size		    9685328 bytes
Variable Size		  654311424 bytes
Database Buffers	  570425344 bytes
Redo Buffers		    7090176 bytes
Database mounted.
Database opened.

CONTAINER: Resetting SYS and SYSTEM passwords.

User altered.


User altered.

############################################
ORACLE PASSWORD FOR SYS AND SYSTEM: MmJiODk1
############################################

#########################
DATABASE IS READY TO USE!
#########################
...
...
Database closed.
Database dismounted.
ORACLE instance shut down.
CONTAINER: stopping container.
[gvenzl@localhost oci-oracle-xe]$ podman volume ls
DRIVER      VOLUME NAME
local       test
[gvenzl@localhost oci-oracle-xe]$
```

The only thing you notice is a delayed startup time, basically the time it takes the runtime to copy whatever is in `/opt/oracle/oradata` into the volume.

However, if you use an OS path, nothing happens, you just mount over the location and everything is gone:

```
[gvenzl@localhost oci-oracle-xe]$ mkdir /tmp/empty
[gvenzl@localhost oci-oracle-xe]$ podman run -ti --name test -v /tmp/empty:/opt/oracle/oradata --entrypoint=bash gvenzl/oracle-xe:21.3.0-slim-faststart
bash-4.4$ ls -al /opt/oracle/oradata
total 0
drwxrwxr-x  2 root   root      6 Dec  3 18:28 .
drwxr-xr-x 11 oracle oinstall 39 Oct 30 04:08 ..
```

You perhaps will be better off using named volumes for the faststart images but in any case, I will highlight this a bit more in the doc, tanks for that suggestion!

**Comment by gvenzl** [comment]
Merged in MAIN
